src/App.tsx

```diff
@@ -1,24 +1,208 @@
+          setError("");
+        }, 3000);
+      })
+      .finally(() => {
+        setTemporaryTodo(undefined);
+      });
+  };
+
+  const removeTodo = (id: number) => {
+    return deleteTodo(id)
+      .then(() => {
+        setTodos(todos.filter((todo) => todo.id !== id));
+      })
+      .catch(() => {
+        setError("Cant to delete a todo");
+        setTimeout(() => {
+          setError("");
+        }, 3000);
+      });
+  };
+  const handleUpdate = async (id: number, data: Partial<Todos>) => {
+    try {
+      await patchTodo(id, data);
+
+      setTodos((state) =>
+        state.map((todo) => {
+          if (todo.id === id) {
+            return { ...todo, ...data };
+          }
+
+          return todo;
+        })
+      );
+    } catch {
+      setError("Unable to add a todo");
+      setTimeout(() => {
+        setError("");
+      }, 3000);
+    }
+  };
+  const toggleAllCompleted = useCallback(() => {
+    const allCompleted = todos.every((todo) => todo.completed);
+
+    if (allCompleted) {
+      todos.map((todoEl) => {
+        return handleUpdate(todoEl.id, { completed: false });
+      });
+    } else {
+      const notCompleted = todos.filter((el) => !el.completed);
+
+      notCompleted.map((todoElement) => {
+        return handleUpdate(todoElement.id, { completed: true });
+      });
+    }
+  }, [todos]);
+
+  const onDeleteComplete = useCallback(() => {
+    const completedTodos = todos.filter((todo) => todo.completed);
+
+    completedTodos.map(async (todo) => {
+      try {
+        await deleteTodo(todo.id);
+        setTodos(todos.filter((task) => !task.completed));
+      } catch {
+        setError("Unable to remove todo");
+        setTimeout(() => {
+          setError("");
+        }, 3000);
+      }
+    });
+  }, [todos]);
+
+  const handleSubmit = (e: { preventDefault: () => void }) => {
+    e.preventDefault();
+    if (!title || title.trim() === "") {
+      setError("Tittle can not be empty");
+      setTimeout(() => {
+        setError("");
+      }, 3000);
+
+      return;
+    }
+
+    addTodo(title);
+    setTitle("");
+  };
+
+  const changeHandler = (e: {
+    target: { value: React.SetStateAction<string> };
+  }) => {
+    setTitle(e.target.value);
+  };
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          onSubmitForm={handleSubmit}
+          title={title}
+          onChangeTitle={changeHandler}
+          onToggleAllCompleted={toggleAllCompleted}
+        />
+
+        {todos.length > 0 && (
```

Comment:
```suggestion
        {todos.length && (
```

Reply:
if I write todos.length its shows  0 when there is no list but in the case  of  todos.length > 0 if there is no list it doesn't show anything